TipCatDotNet.Api/Services/Payments/PaymentService.cs

```diff
@@ -35,39 +44,192 @@ Result Validate()
             }
 
 
-            async Task<Result<PaymentDetailsResponse>> ProceedPayment()
+            async Task<Result<PaymentIntent>> ProceedPayment()
             {
-                //TODO: use payment gateway API for proceed payment
-                return Result.Failure<PaymentDetailsResponse>($"Payment declined.");
+                var amount = Int64.Parse(paymentRequest.TipsAmount.Amount.ToString().Replace(",", ""));
+
+                var createOptions = new PaymentIntentCreateOptions
+                {
+                    PaymentMethodTypes = PaymentEnums.PaymentMethodService.GetAllowed(),
+                    Description = "Tips",
+                    Amount = amount,
+                    Currency = paymentRequest.TipsAmount.Currency.ToString(),
+                    Metadata = new Dictionary<string, string>
+                    {
+                        { "MemberId", paymentRequest.MemberId.ToString() },
+                    },
+                    CaptureMethod = "manual"
+                };
+                try
+                {
+                    var paymentIntent = await _paymentIntentService.CreateAsync(createOptions, cancellationToken: cancellationToken);
+                    return Result.Success(paymentIntent);
+                }
+                catch (StripeException ex)
+                {
+                    return Result.Failure<PaymentIntent>(ex.Message);
+                }
             }
         }
 
 
-        public Task<Result<PaymentDetailsResponse>> GetDetails(string memberCode, CancellationToken cancellationToken = default)
+        public Task<Result<PaymentDetailsResponse>> Capture(string paymentIntentId, CancellationToken cancellationToken = default)
+            => Result.Success()
+                .Bind(() => CapturePayment(paymentIntentId, cancellationToken))
+                .Bind(paymentIntent => GetPaymentDetails(paymentIntent, cancellationToken));
+
+
+        public Task<Result<PaymentDetailsResponse>> GetMemberDetails(string memberCode, CancellationToken cancellationToken = default)
```

Comment:
That're not only a member details

Reply:
There is exactly getting memberDetails without intent details. It's a method needed for getting memberDetails in prepare stage.

Reply:
It will contain account and facility details as well. You may check that out in design mockups

Reply:
Okay